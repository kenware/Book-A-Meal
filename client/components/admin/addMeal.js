import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Dropzone from 'react-dropzone';
import FormData from 'form-data';
import PropTypes from 'prop-types';
import * as mealActions from '../../redux/Action/mealAction';
import * as actions from '../../redux/Action/action';
import './index.scss';

export class Add extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      price: '',
      description: '',
      file: [],
      addMeal: 'Add Meal',
      validDescription: '',
      validPrice: '',
      validName: '',
      image: ''
    };
    this.onChange = this.onChange.bind(this);
    this.addMeal = this.addMeal.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }
  /**
   * lifecycle hook called when component receives props
   *
   * return new state after it is changed
   */
  static getDerivedStateFromProps(props) {
    if (props.errorMessage.createMealError || props.successMessage.createMealSuccess) {
      return { addMeal: 'Add Meal' };
    }
    return null;
  }
  /**
   * remove error props in redux store if component unmount
   */
  componentWillUnmount() {
    this.props.actions.clearMessages();
  }
  /**
   * @param  {} e set state on input onchange event
  */
  onChange(e) {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }
  /**
   * @param  {} file sets image to state using drop box
  */
  onDrop(file) {
    this.setState({ file, image: '' });
  }
  /**
   * calls a redux function
   * adds a meal using form-data
  */
  addMeal(e) {
    e.preventDefault();
    this.props.actions.clearMessages();
    const {
      name, price, description, file
    } = this.state;
    if (!name) { this.setState({ validName: 'Name is required' }); return; }
    if (!price) { this.setState({ validPrice: 'Price is required' }); return; }
    if (!description) { this.setState({ validDescription: 'Description is required' }); return; }
    const payload = new FormData();
    payload.append('name', name);
    payload.append('price', price);
    payload.append('description', description);
    file.forEach((photo) => {
      payload.append('file', photo);
    });
    this.setState({
      addMeal: (<div><i className="fa fa-spinner fa-spin fa-2x fa-fw" aria-hidden="true" /></div>)
    });
    this.props.mealActions.createMeal(payload);
  }
  render() {
    return (
      <div>
        <h2 style={{ marginTop: '2rem' }}>ADD A MEAL </h2>
        <div className="register-container" style={{ marginTop: '2px' }}>
          <div className="register-col" />
          <div className="register-wrapper" style={{ marginTop: '2px' }}>
            <h4 className="p-color text-center">Fill The Field Below To Add A Meal</h4><br />
            <h3 className="text-center danger">{this.props.errorMessage.createMealError }</h3>
            <h3 className="text-center danger">{this.state.message }</h3>
            <form>
              <div className="form-field">
                <label htmlFor="name">Name     <br />
                  <font color="red">{this.state.validName} </font>
                </label>
                <input onChange={this.onChange} type="text" id="name" name="name" placeholder="Jollof Rice" required />
              </div>
              <div className="form-field">
                <label htmlFor="price">Price<br />
                  <font color="red">{this.state.validPrice} </font>
                </label>
                <input onChange={this.onChange} type="number" id="price" name="price" placeholder="5000" required />
              </div>
              <div className="form-field">
                <label htmlFor="name">Description<br />
                  <font color="red">{this.state.validDescription} </font>
                </label>
                <textarea cols="7" rows="9" id="description" onChange={this.onChange} className="form-label descriptions" name="description" placeholder="Description of the meal" required />

              </div>
              <div className="form-field">
                <label htmlFor="password">
                  Image
                </label>
                <span className="form-label">
                  <Dropzone
                    onDrop={this.onDrop}
                    accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
                    ref="dropzone"
                    multiple={false}
                  >
                    Drag and drop or click to select an image to upload.
                  </Dropzone>
                  {this.state.file.map(fil => <img key={fil.preview} src={fil.preview} className="img-fluid" style={{ width: '200px', height: '200px' }} alt="upload" />)}
                </span>
              </div>
              <div className="form-field">
                <button type="submit" className="button lg submit" name="submit" onClick={this.addMeal}>{this.state.addMeal}</button>
              </div>
              <div className="form-field">
                <span className="text-center form-label p-color">{this.props.successMessage.createMealSuccess }</span>
              </div>
            </form>

          </div>
          <div className="register-col" />
        </div>
      </div>
    );
  }
}
Add.propTypes = {
  errorMessage: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  mealActions: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};
export function mapStateToProps(state) {
  return {
    errorMessage: state.errorMessage,
    successMessage: state.successMessage
  };
}
export function mapDispatchToProps(dispatch) {
  return {
    mealActions: bindActionCreators(mealActions, dispatch),
    actions: bindActionCreators(actions, dispatch)
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(Add);
