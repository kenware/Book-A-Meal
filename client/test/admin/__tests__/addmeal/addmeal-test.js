import React from 'react';
import { shallow, mount } from 'enzyme';
import { Add, mapDispatchToProps, mapStateToProps } from '../../../../components/admin/addMeal';
import { props, emptyProps } from '../../adminMockData';

describe('Test index Component of addmeal component', () => {
  it('renders correctly', () => {
    const tree = mount(<Add {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should have one addmeal container class', () => {
    const tree = mount(<Add {...emptyProps} />);
    expect(tree.find('.register-container')).toHaveLength(1);
  });

  it('should have one form tag class', () => {
    const tree = mount(<Add {...emptyProps} />);
    expect(tree.find('form')).toHaveLength(1);
  });

  it('should have four form-field class', () => {
    const tree = mount(<Add {...emptyProps} />);
    expect(tree.find('.form-field')).toHaveLength(6);
  });

  it('should have one addmeal-wrapper', () => {
    const tree = shallow(<Add{...emptyProps} />);
    expect(tree.find('.register-wrapper')).toHaveLength(1);
  });

  it('should have three label', () => {
    const tree = shallow(<Add {...emptyProps} />);
    expect(tree.find('label')).toHaveLength(4);
  });

  it('testing initial states', () => {
    const tree = shallow(<Add {...emptyProps} />);
    expect(tree.state('name')).toEqual('');
    expect(tree.state('price')).toEqual('');
    expect(tree.state('description')).toEqual('');
    expect(tree.state('addMeal')).toEqual('Add Meal');
    expect(tree.state('file')).toEqual([]);
    expect(tree.state('image')).toEqual('');
  });

  it('it should rerender component on new props receipt', () => {
    let tree = shallow(<Add {...emptyProps} />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<Add {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to change event and change the state of the name input', () => {
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('#name').simulate('change', { target: { name: 'name', value: 'rice' } });
    expect(tree.state('name')).toEqual('rice');
  });

  it('should respond to change event and change the state of the price input', () => {
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('#price').simulate('change', { target: { name: 'price', value: '100' } });
    expect(tree.state('price')).toEqual('100');
  });

  it('should respond to change event and change the state of the description input', () => {
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('.descriptions').simulate('change', { target: { name: 'description', value: 'very delicious' } });
    expect(tree.state('description')).toEqual('very delicious');
  });

  it('should respond to addMeal onClick method ', () => {
    const addMeal = jest.spyOn(Add.prototype, 'addMeal');
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('.submit').simulate('click', { target: { name: 'submit' }, preventDefault: jest.fn() });
    expect(addMeal).toHaveBeenCalled();
  });

  it('should respond to addMeal onClick method and return error when name input is not supplied', () => {
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('.submit').simulate('click', { target: { name: 'submit' }, preventDefault: jest.fn() });
    expect(tree.state('validName')).toEqual('Name is required');
  });

  it('should respond addMeal method onClick event and return error when price input is empty', () => {
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('#name').simulate('change', { target: { name: 'name', value: 'rice' } });
    tree.find('.submit').simulate('click', { target: { name: 'submit' }, preventDefault: jest.fn() });
    expect(tree.state('validPrice')).toEqual('Price is required');
  });

  it('should respond addMeal method onClick event and return error when description input is empty', () => {
    const tree = shallow(<Add {...emptyProps} />);
    tree.find('#name').simulate('change', { target: { name: 'name', value: 'rice' } });
    tree.find('#price').simulate('change', { target: { name: 'price', value: '100' } });
    tree.find('.submit').simulate('click', { target: { name: 'submit' }, preventDefault: jest.fn() });
    expect(tree.state('validDescription')).toEqual('Description is required');
    tree.find('#description').simulate('change', { target: { name: 'description', value: 'good' } });
    tree.find('.submit').simulate('click', { target: { name: 'submit' }, preventDefault: jest.fn() });
    expect(tree).toMatchSnapshot();
  });

  it('should should respond to lifeCycle Methods', () => {
    const tree = shallow(<Add {...props} />);
    const componentWillReceiveProps = jest.spyOn(Add.prototype, 'componentWillReceiveProps');
    const componentWillUnmount = jest.spyOn(Add.prototype, 'componentWillUnmount');

    const wrapper = tree.instance();
    wrapper.componentWillReceiveProps({
      errorMessage: { createMealError: '' },
      successMessage: { createMealSuccess: '' }
    });
    wrapper.componentWillUnmount();
    expect(tree).toMatchSnapshot();
    tree.instance().componentWillReceiveProps({
      errorMessage: { createMealError: 'error' },
      successMessage: { createMealSuccess: 'success' }
    });
    expect(tree).toMatchSnapshot();
    expect(componentWillReceiveProps).toHaveBeenCalled();
    expect(componentWillUnmount).toMatchSnapshot();
  });

  it('should should respond to dropBox onDRop Methods', () => {
    const tree = shallow(<Add {...props} />);
    const wrapper = tree.instance();
    wrapper.onDrop([{ preview: 'image/l.jpg' }]);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to mapStateToProps methods', () => {
    const ownProps = { match: { params: { mealId: 1 } } };
    const tree = mapStateToProps(props, ownProps);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to mapDispatchToProps methods', () => {
    const tree = mapDispatchToProps(emptyProps);
    expect(tree).toMatchSnapshot();
  });
});
