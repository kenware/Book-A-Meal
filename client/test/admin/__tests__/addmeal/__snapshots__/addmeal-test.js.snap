// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test index Component of addmeal component it should change display error message from redux error message props 1`] = `
<div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    ADD A MEAL 
  </h2>
  <div
    className="register-container"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <h4
        className="p-color text-center"
      >
        Fill The Field Below To Add A Meal
      </h4>
      <br />
      <h3
        className="text-center danger"
      />
      <h3
        className="text-center danger"
      />
      <form>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Name     
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Jollof Rice"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="price"
          >
            Price
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="price"
            name="price"
            onChange={[Function]}
            placeholder="5000"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Description
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <textarea
            className="form-label descriptions"
            cols="7"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description of the meal"
            required={true}
            rows="9"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Image
          </label>
          <span
            className="form-label"
          >
            <t
              accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={false}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              Drag and drop or click to select an image to upload.
            </t>
          </span>
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg submit"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Add Meal
          </button>
        </div>
        <div
          className="form-field"
        >
          <span
            className="text-center form-label p-color"
          />
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
</div>
`;

exports[`Test index Component of addmeal component it should change display error message from redux error message props 2`] = `
<div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    ADD A MEAL 
  </h2>
  <div
    className="register-container"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <h4
        className="p-color text-center"
      >
        Fill The Field Below To Add A Meal
      </h4>
      <br />
      <h3
        className="text-center danger"
      >
        Wrong credentials
      </h3>
      <h3
        className="text-center danger"
      />
      <form>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Name     
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Jollof Rice"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="price"
          >
            Price
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="price"
            name="price"
            onChange={[Function]}
            placeholder="5000"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Description
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <textarea
            className="form-label descriptions"
            cols="7"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description of the meal"
            required={true}
            rows="9"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Image
          </label>
          <span
            className="form-label"
          >
            <t
              accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={false}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              Drag and drop or click to select an image to upload.
            </t>
          </span>
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg submit"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Add Meal
          </button>
        </div>
        <div
          className="form-field"
        >
          <span
            className="text-center form-label p-color"
          >
            meal created
          </span>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
</div>
`;

exports[`Test index Component of addmeal component renders correctly 1`] = `
<Add
  actions={
    Object {
      "clearMessages": [MockFunction],
      "deleteMeal": [MockFunction],
      "refreshToken": [MockFunction],
    }
  }
  activePage={1}
  allOrder={
    Object {
      "orders": Array [
        Object {
          "address": "Reverend Street",
          "createdAt": "2018-05-4",
          "id": 1,
          "meals": Array [
            Object {
              "descrption": "Very delicious",
              "id": 1,
              "name": "rice",
              "orderMealItems": Object {
                "quantity": 3,
                "totalPrice": 4,
              },
              "price": "200",
            },
          ],
          "quantity": 3,
          "status": "confirmed",
          "updatedAt": "2018-05-4",
          "user": Object {
            "id": 1,
            "image": "image",
            "name": "kevin",
          },
        },
      ],
    }
  }
  errorMessage={
    Object {
      "createMealError": "Wrong credentials",
    }
  }
  handlePageChange={[MockFunction]}
  mealActions={
    Object {
      "deleteMeal": [MockFunction],
      "getAllMeals": [MockFunction],
      "loadMostOrderedMeal": [MockFunction],
    }
  }
  meals={
    Object {
      "rows": Array [
        Object {
          "descrption": "Very delicious",
          "id": 1,
          "name": "rice",
          "price": "200",
        },
        Object {
          "descrption": "Very delicious",
          "id": 2,
          "name": "beans",
          "price": "200",
        },
      ],
    }
  }
  menuActions={
    Object {
      "getAllOrders": [MockFunction],
      "setMenu": [MockFunction],
    }
  }
  mostOrder={
    Array [
      Object {
        "description": "good",
        "id": 1,
        "image": "image",
        "name": "rice",
        "orderCount": 4,
        "price": "400",
      },
    ]
  }
  notific={Array []}
  orderActions={
    Object {
      "getAllOrders": [MockFunction],
    }
  }
  successMessage={
    Object {
      "createMealSuccess": "meal created",
    }
  }
>
  <div>
    <h2
      style={
        Object {
          "marginTop": "2rem",
        }
      }
    >
      ADD A MEAL 
    </h2>
    <div
      className="register-container"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <div
        className="register-col"
      />
      <div
        className="register-wrapper"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <h4
          className="p-color text-center"
        >
          Fill The Field Below To Add A Meal
        </h4>
        <br />
        <h3
          className="text-center danger"
        >
          Wrong credentials
        </h3>
        <h3
          className="text-center danger"
        />
        <form>
          <div
            className="form-field"
          >
            <label
              htmlFor="name"
            >
              Name     
              <br />
              <font
                color="red"
              >
                 
              </font>
            </label>
            <input
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Jollof Rice"
              required={true}
              type="text"
            />
          </div>
          <div
            className="form-field"
          >
            <label
              htmlFor="price"
            >
              Price
              <br />
              <font
                color="red"
              >
                 
              </font>
            </label>
            <input
              id="price"
              name="price"
              onChange={[Function]}
              placeholder="5000"
              required={true}
              type="number"
            />
          </div>
          <div
            className="form-field"
          >
            <label
              htmlFor="name"
            >
              Description
              <br />
              <font
                color="red"
              >
                 
              </font>
            </label>
            <textarea
              className="form-label descriptions"
              cols="7"
              id="description"
              name="description"
              onChange={[Function]}
              placeholder="Description of the meal"
              required={true}
              rows="9"
            />
          </div>
          <div
            className="form-field"
          >
            <label
              htmlFor="password"
            >
              Image
            </label>
            <span
              className="form-label"
            >
              <t
                accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
                disableClick={false}
                disablePreview={false}
                disabled={false}
                maxSize={Infinity}
                minSize={0}
                multiple={false}
                onDrop={[Function]}
                preventDropOnDocument={true}
              >
                <div
                  aria-disabled={false}
                  className=""
                  onClick={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                  onDragOver={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  style={
                    Object {
                      "borderColor": "#666",
                      "borderRadius": 5,
                      "borderStyle": "dashed",
                      "borderWidth": 2,
                      "height": 200,
                      "width": 200,
                    }
                  }
                >
                  Drag and drop or click to select an image to upload.
                  <input
                    accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
                    autoComplete="off"
                    disabled={false}
                    multiple={false}
                    onChange={[Function]}
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="file"
                  />
                </div>
              </t>
            </span>
          </div>
          <div
            className="form-field"
          >
            <button
              className="button lg submit"
              name="submit"
              onClick={[Function]}
              type="submit"
            >
              Add Meal
            </button>
          </div>
          <div
            className="form-field"
          >
            <span
              className="text-center form-label p-color"
            >
              meal created
            </span>
          </div>
        </form>
      </div>
      <div
        className="register-col"
      />
    </div>
  </div>
</Add>
`;

exports[`Test index Component of addmeal component should respond addMeal method onClick event and return error when description input is empty 1`] = `
<div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    ADD A MEAL 
  </h2>
  <div
    className="register-container"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <h4
        className="p-color text-center"
      >
        Fill The Field Below To Add A Meal
      </h4>
      <br />
      <h3
        className="text-center danger"
      />
      <h3
        className="text-center danger"
      />
      <form>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Name     
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Jollof Rice"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="price"
          >
            Price
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="price"
            name="price"
            onChange={[Function]}
            placeholder="5000"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Description
            <br />
            <font
              color="red"
            >
              Description is required
               
            </font>
          </label>
          <textarea
            className="form-label descriptions"
            cols="7"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description of the meal"
            required={true}
            rows="9"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Image
          </label>
          <span
            className="form-label"
          >
            <t
              accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={false}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              Drag and drop or click to select an image to upload.
            </t>
          </span>
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg submit"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            <div>
              <i
                aria-hidden="true"
                className="fa fa-spinner fa-spin fa-2x fa-fw"
              />
            </div>
          </button>
        </div>
        <div
          className="form-field"
        >
          <span
            className="text-center form-label p-color"
          />
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
</div>
`;

exports[`Test index Component of addmeal component should respond to mapDispatchToProps methods 1`] = `
Object {
  "actions": Object {
    "changePassword": [Function],
    "clearMessages": [Function],
    "getNotifications": [Function],
    "getUser": [Function],
    "loadErrorMessage": [Function],
    "loadNotifications": [Function],
    "loadSuccessMessage": [Function],
    "loadUser": [Function],
    "login": [Function],
    "refreshToken": [Function],
    "register": [Function],
    "resetLink": [Function],
    "updateProfile": [Function],
    "upgrade": [Function],
  },
  "mealActions": Object {
    "createMeal": [Function],
    "deleteMeal": [Function],
    "getAllMeals": [Function],
    "loadAllMeals": [Function],
    "loadErrorMessage": [Function],
    "loadMostOrdered": [Function],
    "loadMostOrderedMeal": [Function],
    "loadSuccessMessage": [Function],
    "updateMeal": [Function],
  },
}
`;

exports[`Test index Component of addmeal component should respond to mapStateToProps methods 1`] = `
Object {
  "errorMessage": Object {
    "createMealError": "Wrong credentials",
  },
  "successMessage": Object {
    "createMealSuccess": "meal created",
  },
}
`;

exports[`Test index Component of addmeal component should should respond to dropBox onDRop Methods 1`] = `
<div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    ADD A MEAL 
  </h2>
  <div
    className="register-container"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <h4
        className="p-color text-center"
      >
        Fill The Field Below To Add A Meal
      </h4>
      <br />
      <h3
        className="text-center danger"
      >
        Wrong credentials
      </h3>
      <h3
        className="text-center danger"
      />
      <form>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Name     
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Jollof Rice"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="price"
          >
            Price
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="price"
            name="price"
            onChange={[Function]}
            placeholder="5000"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Description
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <textarea
            className="form-label descriptions"
            cols="7"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description of the meal"
            required={true}
            rows="9"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Image
          </label>
          <span
            className="form-label"
          >
            <t
              accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={false}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              Drag and drop or click to select an image to upload.
            </t>
          </span>
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg submit"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Add Meal
          </button>
        </div>
        <div
          className="form-field"
        >
          <span
            className="text-center form-label p-color"
          >
            meal created
          </span>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
</div>
`;

exports[`Test index Component of addmeal component should should respond to lifeCycle Methods 1`] = `
<div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    ADD A MEAL 
  </h2>
  <div
    className="register-container"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <h4
        className="p-color text-center"
      >
        Fill The Field Below To Add A Meal
      </h4>
      <br />
      <h3
        className="text-center danger"
      >
        Wrong credentials
      </h3>
      <h3
        className="text-center danger"
      />
      <form>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Name     
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Jollof Rice"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="price"
          >
            Price
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="price"
            name="price"
            onChange={[Function]}
            placeholder="5000"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Description
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <textarea
            className="form-label descriptions"
            cols="7"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description of the meal"
            required={true}
            rows="9"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Image
          </label>
          <span
            className="form-label"
          >
            <t
              accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={false}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              Drag and drop or click to select an image to upload.
            </t>
          </span>
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg submit"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Add Meal
          </button>
        </div>
        <div
          className="form-field"
        >
          <span
            className="text-center form-label p-color"
          >
            meal created
          </span>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
</div>
`;

exports[`Test index Component of addmeal component should should respond to lifeCycle Methods 2`] = `
<div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    ADD A MEAL 
  </h2>
  <div
    className="register-container"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
      style={
        Object {
          "marginTop": "2px",
        }
      }
    >
      <h4
        className="p-color text-center"
      >
        Fill The Field Below To Add A Meal
      </h4>
      <br />
      <h3
        className="text-center danger"
      >
        Wrong credentials
      </h3>
      <h3
        className="text-center danger"
      />
      <form>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Name     
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Jollof Rice"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="price"
          >
            Price
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <input
            id="price"
            name="price"
            onChange={[Function]}
            placeholder="5000"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Description
            <br />
            <font
              color="red"
            >
               
            </font>
          </label>
          <textarea
            className="form-label descriptions"
            cols="7"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description of the meal"
            required={true}
            rows="9"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Image
          </label>
          <span
            className="form-label"
          >
            <t
              accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={false}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              Drag and drop or click to select an image to upload.
            </t>
          </span>
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg submit"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Add Meal
          </button>
        </div>
        <div
          className="form-field"
        >
          <span
            className="text-center form-label p-color"
          >
            meal created
          </span>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
</div>
`;
